zulip/zulip/__init__.py

```diff
@@ -601,8 +601,16 @@ def call_on_each_event(
         if narrow is None:
             narrow = []
 
-        def do_register() -> Tuple[str, int]:
-            while True:
+        queue_id = None
+        # NOTE: Back off exponentially to cover against potential bugs in this
+        #       library causing a DoS attack against a server when getting errors
+        #       (explicit values listed for clarity)
+        backoff = RandomExponentialBackoff(maximum_retries=10,
+                                           timeout_success_equivalent=300,
```

Comment:
I don't think we want a `timeout_success_equivalent` on this.  That's for cases where we don't have the ability to put an explicit `success()` call in, and we want to be able to reset our backoff after 5 minutes on the assumption that that's a "success."  Here, we have places we can mark as successful.  Having a `timeout_success_equivalent` muddies the logic -- and I think won't kick in unless we have 5-minute HTTP requests, which should really not be treated as successes. :)